### PR TITLE
ci: add Terraform logging

### PR DESCRIPTION
**Pull request by msanft**
### Proposed change(s)

- Upload a Terraform log artifact at `DEBUG` level for manual inspection of Terraform failures.
- [AWS Run](https://github.com/edgelesssys/constellation/actions/runs/4718634933)
- [Azure Run](https://github.com/edgelesssys/constellation/actions/runs/4718638266)
- [GCP Run](https://github.com/edgelesssys/constellation/actions/runs/4718637100)

### Additional info
- Change to `JSON` level later on and upload the logs to OpenSearch.

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

File: .github/actions/upload_terraform_logs/action.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+description: "Upload the Terraform log file as an artifact."
+
+runs:
+  using: "composite"
+  steps:
+    - name: Upload artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: terraform-logs
+        path: terraform.log
```
Comment (katexochen) [inline .github/actions/upload_terraform_logs/action.yml:11]: We should add the prefix from the e2e_test as suffix to the filename.

Comment (katexochen) [reply .github/actions/upload_terraform_logs/action.yml:11]: Thats what I meant: https://github.com/edgelesssys/constellation/blob/main/.github/actions/e2e_test/action.yml#L170-L177

File: .github/workflows/e2e-test-weekly.yml [hunk 2]
```diff
@@ -241,6 +241,12 @@ jobs:
           kubernetesVersion: ${{ matrix.kubernetes-version }}
           provider: ${{ matrix.provider }}
 
+      - name: Always upload Terraform logs
+        if: always()
+        uses: ./.github/actions/upload_terraform_logs
+        with:
+          artifactNameSuffix: ${{ steps.e2e_test.outputs.namePrefix }}
```
Comment (katexochen) [inline .github/workflows/e2e-test-weekly.yml:248]: Can't we do this step on a lower layer, like in constellation_destroy? Otherwise lgtm.

**Review by katexochen — APPROVED** [review]

File: .github/workflows/e2e-test-weekly.yml (hunk 2 above)
Comment (msanft) [reply .github/workflows/e2e-test-weekly.yml:248]: I think it'd be a bad practice to put it in either constellation_destroy or constellation_iam_destroy. If you put it in 1 of the 2 you would implicitly require that action to be always executed last (to not lose log output)

**Pull request merged**